sn_networking/src/cmd.rs

```diff
@@ -538,12 +538,15 @@ impl SwarmDriver {
                 // be handled.
                 // `self` then handles the request and sends a response back again to itself.
                 if peer == *self.swarm.local_peer_id() {
-                    trace!("Sending request to self");
-
-                    self.send_event(NetworkEvent::RequestReceived {
-                        req,
-                        channel: MsgResponder::FromSelf(sender),
-                    });
+                    trace!("Sending query request to self");
+                    if let Request::Query(query) = req {
+                        self.send_event(NetworkEvent::QueryRequestReceived {
+                            query,
+                            channel: MsgResponder::FromSelf(sender),
+                        });
+                    } else {
+                        trace!("Replicate cmd to self received, ignoring");
```

Comment:
the log is bit confusing.
As I understand, it is a `non-query request to self, which can be ignored`, maybe due to `replicate` ?

Reply:
ah yeh, we'd be asking ourself to replicate data which we're already holding? So can be ignored